patches/minecraft/net/minecraft/client/renderer/chunk/RenderChunk.java.patch

```diff
@@ -1,6 +1,17 @@
 --- ../src-base/minecraft/net/minecraft/client/renderer/chunk/RenderChunk.java
 +++ ../src-work/minecraft/net/minecraft/client/renderer/chunk/RenderChunk.java
-@@ -162,7 +162,7 @@
+@@ -157,12 +157,18 @@
+                 IBlockState iblockstate = this.field_189564_r.func_180495_p(blockpos$mutableblockpos);
+                 Block block = iblockstate.func_177230_c();
+ 
++                if (this.field_189564_r.func_175624_G() != net.minecraft.world.WorldType.field_180272_g)
++                {
++                    iblockstate = iblockstate.func_185899_b(this.field_189564_r, blockpos$mutableblockpos);
++                    iblockstate = block.getExtendedState(iblockstate, this.field_189564_r, blockpos$mutableblockpos);
```

Comment:
Again this would call these functions in the bulk of unneeded cases. To do it properly performance wise without extra side effects, you would have to just change the canRenderInLayer call below. We add that function so it'd be better to delay any getting of extra data to opt-in mods.